### PR TITLE
feat: custom index route

### LINKED ISSUE #1285: [FEAT] Manually manage which page the application opens on by default
**Is your feature request related to a problem? Please describe.**
I want to use two dashboard in my app so I didin't pass the Dashboard prop to `<Refine>` component. Resulting of that i couldn't manage which page is my opening default page. When I go to the "/" url, **refine** redirects to the first route of the resources but i want to go specific url like "dashboard/analysis". 

I would like to manually manage which page the application opens on by default.

**Describe the solution you'd like**
I want to manage which route is my default opening page in app.

### PR DESCRIPTION
**Pull request by aliemir**
Added `indexRoute` property to `options` of `<Refine/>` component to manage initial route of the app if users wants to change that from first resource's route to something else.

### Test plan (required)

Passing other than a `react-query` issue in the `next`.

### Closing issues

Resolves #1285 #2544 #2545 #2546

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: packages/core/src/hooks/auth/useLogin/index.ts [hunk 1]
```diff
@@ -47,7 +54,7 @@ export const useLogin = <TVariables = {}>(): UseMutationResult<
                     if (typeof redirectPathFromAuth === "string") {
                         replace(redirectPathFromAuth);
                     } else {
-                        replace("/");
+                        replace(options.indexRoute ? options.indexRoute : "/");
```
Comment (omeraplak) [inline packages/core/src/hooks/auth/useLogin/index.ts:57]:
```suggestion
                        replace(options.indexRoute ?? "/");
```

**Comment by omeraplak** [comment]
🎖️

**Review by biskuvit — APPROVED** [review]

File: documentation/docs/core/components/refine-config.md [hunk 2]
```diff
@@ -438,6 +438,29 @@ If you don't have a show page and you redirect to the show page, the user will b
...
 
+### `indexRoute`
+
+By default, **refine** redirects to the `list` page of the first resource. To change this behaviour, set `indexRoute` as follows:
+
+```tsx title="App.tsx"
+const App: React.FC = () => {
+    return (
+        <Refine
+            {/* ... */}
```
Comment (salihozdemir) [inline documentation/docs/core/components/refine-config.md:449]:
```suggestion
            ...
```

File: packages/core/src/hooks/auth/useLogin/index.ts (hunk 1 above)
Comment (salihozdemir) [reply packages/core/src/hooks/auth/useLogin/index.ts:57]: I think, we can set the default value of the indexRoute in RefineContext thus we don't need to any checking in here.

File: packages/core/src/contexts/refine/index.tsx [hunk 3]
```diff
@@ -15,6 +15,7 @@ export const defaultRefineOptions: IRefineContextOptions = {
     undoableTimeout: 5000,
     warnWhenUnsavedChanges: false,
     liveMode: "off",
+    indexRoute: "",
```
Comment (salihozdemir) [inline packages/core/src/contexts/refine/index.tsx:18]:
```suggestion
    indexRoute: "/",
```

I think, we can manage the refine default values in here.

File: documentation/docs/core/components/refine-config.md [hunk 4]
```diff
@@ -438,6 +438,29 @@ If you don't have a show page and you redirect to the show page, the user will b
 
 :::
 
+### `indexRoute`
```
Comment (omeraplak) [inline documentation/docs/core/components/refine-config.md:441]: How about the idea of getting the routing-related options via the `routerProvider`? like custom routes

**Comment by salihozdemir** [comment]
How is it if we rename "openingRoute", "landingPage" or "startupPage" instead of "indexRoute". "indexRoute" name doesn't  explain itself well enough.

**Comment by aliemir** [comment]
We'll move the option to `routerProvider` functions and rename it to `defaultRoute` if everyone agrees.

**Review by omeraplak — APPROVED** [review]

File: .changeset/gorgeous-keys-relax.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@pankod/refine-remix-router": minor
+---
+
+Add splat rotue support to remix with `handleRefineParams` helper.
```
Comment (omeraplak) [inline .changeset/gorgeous-keys-relax.md:5]:
```suggestion
Add splat route support to remix with `handleRefineParams` helper.
```

**Review by yildirayunlu — APPROVED** [review]

**Review by salihozdemir — APPROVED** [review]

**Pull request merged**